### PR TITLE
strip out any \r values hanging on the end of the destination filename

### PR DESCRIPTION
**Pull request by billrawlinson**
tldr; blackbox_postdeploy leaves carriage return at end of unencrypted file name on windows using MinGW

full story:

I'm using windows, mingw and have everything working with blackbox except for blackbox_postdeploy which works - but munges the decrypted file name.

Here is a contrived example.

Let's say I have the files, `secrets1.txt` and `secrets2.text` that I have encrypted using `blackbox_register_file`.  If I use `blackbox_edit secrets1.txt.gpg` the file written to disk is titled `secrets1.txt` as expected.  Perfect!

However, if I use `blackbox_postdeploy` I get a carriage return `\r` at the end of each file generated which doesn't work so well:

`secrets1.txt'$'\r`
`secrets2.txt'$'\r`

Interestingly, if I decrypt each file individually and then use `blackbox_shred_all_files` the file names are processed properly (the \r doesn't seem to manifest) and the correct files are deleted.  

If I use `blackbox_postdeploy`  and then `blackbox_shred_all_files` the shred fails to delete the files because it doesn't use the `\r` in the filename.

becuase this only happens in blackbox_postdeploy I only added the logic in there instead of creating a new function in _blackbox_common.sh

### CODE REVIEW

**Comment by billrawlinson** [comment]
I may have inadvertently caused this issue now that I'm thinking more about it.

I spent a lot of time yesterday trying to make sure everything worked and I may have opened the file "blackbox-files.txt" which would have translated the carriage returns to windows cr/lf at that time.  I may have saved it (I don't recall) which would have injected the extra character into the file.


I have tried the blackbox_postdeploy on a different machine (windows 12) and it worked fine.  My development machine is windows 7.

I'm not really sure but, this "cleaning" doesn't seem to hurt anything but it does help prevent a problem if anyone else opens/saves/closes the file without realizing they may have actually modified the contents.

**Comment by tlimoncelli** [comment]
It sounds like the problem is that an \r is being added to the blackbox-files.txt file.  I'd rather make sure that blackbox_register_new_file doesn't add the \r than fix it afterwords.

Can you run `od -c blackbox-files.txt` and see if there are `\r` bytes being included?

The problem might be in `add_filename_to_cryptlist` in `_blackbox_common.sh`.

Tom

P.S. I really appreciate the help making this work on MinGW!

**Comment by billrawlinson** [comment]
I ran `od -c blackbox-files.txt`  that is an interesting command.

There are some `\r` values in there.  Like I said, I may have caused them by opening the file yesterday while trying to get things working.  

So, know that I know about this command I decided to test my theory that I introduced the `\r` yesterday while trying to figure everything out.

I just registered a new file then ran the `od` command again and there is no `\r` after that one; just after the two that were in there from yesterday.

I probably did introduce it by opening/saving/closing the file yesterday without realizing the consequences of that action.

**Comment by tlimoncelli** [comment]
I'd rather detect that the file has gotten corrupted and error out with a message explaining how to fix the problem.

**Comment by billrawlinson** [comment]
okay

**Comment by billrawlinson** [comment]
I think I figured out how the funky characters are getting in the file.  Many windows git clients will check out files with windows newlines but push them back to github with unix style.

**Comment by billrawlinson** [comment]
I am back to suggesting this pull request as is since the "corruption" isn't really a user introduced problem but rather a natural consequence of using git on windows.

**Comment by tlimoncelli** [comment]
I think a better fix for this is 9ff5892ddfdfb5181a47ad1bad70e0387e6e7db1.  Now a .gitattributes files is created which tells Git not to alter the line endings on any of the administrative files. This should work under MinGW, Bash for Windows, Git for Windows, etc.
